src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorLSPTextDocumentCreatedListener.cs

```diff
@@ -171,6 +183,11 @@ private void InitializeRazorLSPTextBuffer(ITextBuffer textBuffer)
 
                 // We still change the content type for remote scenarios in order to enable our TextMate grammar to light up the Razor editor properly.
                 textBuffer.ChangeContentType(_razorLSPContentType, editTag: null);
+
+                // Temporary: The guest needs to react to the host manually applying edits and moving the cursor.
+                // This can be removed once the client starts supporting snippets.
+                textBuffer.Properties.AddProperty(FilePathPropertyKey, filePath);
+                textBuffer.ChangedHighPriority += RazorGuestBuffer_Changed;
```

Comment:
Interesting, do we not need to unset this on text document in the `TextDocumentFactory_TextDocumentDisposed` method?

Reply:
I assumed it would die along with the TextBuffer. Does the TextBuffer persist even after the document is disposed?

Reply:
Possibly? I'm not entirely sure though. I'd detatch just to be certain